llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp

```diff
@@ -1575,8 +1575,9 @@ static void translateESIMDIntrinsicCall(CallInst &CI) {
     IRBuilder<> Builder(&CI);
 
     NewInst = Builder.CreateStore(
-        NewCI, Builder.CreateBitCast(CastInstruction->getPointerOperand(),
-                                     NewCI->getType()->getPointerTo()));
+        NewCI,
+        Builder.CreateBitCast(CastInstruction->getPointerOperand(),
+                              llvm::PointerType::getUnqual(CI.getContext())));
```

Comment:
Shouldn't be `NewCI->getContext()` instead of `(CI.getContext()` ? Is there a reason for this change?

Reply:
probably the context object is the same but yeah i guess lets be safe

Reply:
I think, it should be even possible to skip the bitcast instruction, and pass the `CastInstruction->getPointerOperand()` directly, since the pointers are opaque